job-simulation.rb

```diff
@@ -1,4 +1,47 @@
 require './Stack.rb'
 require './Queue.rb'
 
+class Company
+  MAX_EMPLOYEES = 6
 
+  attr_reader :waiting, :working
+
+  def initialize
+    @waiting = Queue.new
+    10.times { |i| wait! i+1 }
+
+    @working = Stack.new
+
+    end_of_the_month
+  end
+
+  def end_of_the_month
+    fire!
+    hire!
+  end
+
+  private
+
+  def hire!
+    until working.size >= MAX_EMPLOYEES
+      new_hire = waiting.dequeue
+      working.push new_hire
+    end
+  end
+
+  def fire!
+    return if working.empty?
```

Comment:
I'd have to write a test to know for sure. :) But I don't think so, given the limited interface of the class. Only the `end_of_the_month` method is public, though, so unless folks reach into the private methods (and I'm sure that would _never_ happen), `working` will always have 10 elements when `fire!` is invoked, _except_ the very first time it's called during object initialization. I could likely refactor the guard out if I called `hire!` directly during initialize instead of `end_of_the_month`, but I'd want tests in place before doing so.